src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -747,5 +777,88 @@ void ClusterTaskManager::RemoveFromBacklogTracker(const Task &task) {
   }
 }
 
+void ClusterTaskManager::FreeLocalTaskResources(std::shared_ptr<WorkerInterface> worker) {
+  cluster_resource_scheduler_->FreeLocalTaskResources(worker->GetAllocatedInstances());
+  worker->ClearAllocatedInstances();
+  cluster_resource_scheduler_->FreeLocalTaskResources(
+      worker->GetLifetimeAllocatedInstances());
+  worker->ClearLifetimeAllocatedInstances();
+}
+
+bool ClusterTaskManager::ReleaseCpuResourcesAndMarkWorkerAsBlocked(
+    std::shared_ptr<WorkerInterface> worker, bool release_resources) {
+  if (!worker || worker->IsBlocked() || !release_resources) {
+    return false;
+  }
+  std::vector<double> cpu_instances;
+  if (worker->GetAllocatedInstances() != nullptr) {
+    cpu_instances = worker->GetAllocatedInstances()->GetCPUInstancesDouble();
+  }
+  if (cpu_instances.size() > 0) {
+    std::vector<double> overflow_cpu_instances =
+        cluster_resource_scheduler_->AddCPUResourceInstances(cpu_instances);
+    for (unsigned int i = 0; i < overflow_cpu_instances.size(); i++) {
+      RAY_CHECK(overflow_cpu_instances[i] == 0) << "Should not be overflow";
+    }
+    worker->MarkBlocked();
+  }
+
+  return true;
+}
+
+bool ClusterTaskManager::ReturnCpuResourcesAndMarkWorkerAsUnblocked(
+    std::shared_ptr<WorkerInterface> worker) {
+  // Important: avoid double unblocking if the unblock RPC finishes after task end.
+  if (!worker || !worker->IsBlocked()) {
+    return false;
+  }
+  std::vector<double> cpu_instances;
+  if (worker->GetAllocatedInstances() != nullptr) {
+    cpu_instances = worker->GetAllocatedInstances()->GetCPUInstancesDouble();
+  }
+  if (cpu_instances.size() > 0) {
+    // Important: we allow going negative here, since otherwise you can use infinite
+    // CPU resources by repeatedly blocking / unblocking a task. By allowing it to go
+    // negative, at most one task can "borrow" this worker's resources.
+    cluster_resource_scheduler_->SubtractCPUResourceInstances(
+        cpu_instances, /*allow_going_negative=*/true);
+    worker->MarkUnblocked();
+  }
+  return true;
+}
+
+void ClusterTaskManager::ScheduleAndDispatchTasks() {
+  SchedulePendingTasks();
```

Comment:
The unit test inside `cluster_task_manager_test.cc` will call these method directly.
How about make the methods you mentioned above virtual and mock a ClsuterTaskManager inside `cluster_task_manager_test.cc`? @stephanie-wang

Reply:
It'd be best if you could modify the unit tests to call the new interface methods instead (otherwise we're not really getting proper test coverage because the unit tests won't match the actual usage of the ClusterTaskManager class).

Reply:
Okey, I will fix it.

Reply:
- I've tried to abstract interfaces to `clusters_resource_scheduler_`. However, it is difficult to abstract because the new schedule relies on `TaskResourceInstances`, while the old schedule relies on `ResourceSet`.
- In addition, the old schedule also relies on `local_ queues_` which does not exist in the new scheduler.
cc @raulchen @stephanie-wang